packages/react/src/components/NumberInput/NumberInput.js

```diff
@@ -26,6 +26,345 @@ const defaultTranslations = {
+                    aria-label={decrementNumLabel || iconDescription}
+                    aria-live="polite"
+                    aria-atomic="true">
+                    <CaretDownGlyph className="down-icon" />
+                  </button>
+                </div>
+              </div>
+            </>
+          );
+        })()}
+        {error}
+      </div>
+    </div>
+  );
+});
+
+NumberInputView.propTypes = {
+  /**
+   * Specify an optional className to be applied to the wrapper node
+   */
+  className: PropTypes.string,
+
+  /**
+   * Specify if the control should be disabled, or not
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * Specify whether you want the underlying label to be visually hidden
+   */
+  hideLabel: PropTypes.bool,
+
+  /**
+   * Provide a description for up/down icons that can be read by screen readers
+   */
+  iconDescription: PropTypes.string.isRequired,
+
+  /**
+   * Specify a custom `id` for the input
+   */
+  id: PropTypes.string.isRequired,
+
+  /**
+   * Generic `label` that will be used as the textual representation of what
+   * this field is for
+   */
+  label: PropTypes.node,
+
+  /**
+   * The maximum value.
+   */
+  max: PropTypes.number,
+
+  /**
+   * The minimum value.
+   */
+  min: PropTypes.number,
+
+  /**
+   * Provide an optional function to be called when the value is changed
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Provide an optional function to be called when the up/down button is clicked
+   */
+  onClick: PropTypes.func,
+
+  /**
+   * Specify how much the valus should increase/decrease upon clicking on up/down button
+   */
+  step: PropTypes.number,
+
+  /**
+   * Specify the value of the input
+   */
+  value: PropTypes.number,
+
+  /**
+   * Specify if the currently value is invalid.
+   */
+  invalid: PropTypes.bool,
+
+  /**
+   * Message which is displayed if the value is invalid.
+   */
+  invalidText: PropTypes.string,
+
+  /**
+   * Provide text that is used alongside the control label for additional help
+   */
+  helperText: PropTypes.node,
+
+  /**
+   * Provide a description that would be used to best describe the use case of the NumberInput component
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * `true` to use the light version.
+   */
+  light: PropTypes.bool,
+
+  /**
+   * `true` to allow empty string.
+   */
+  allowEmpty: PropTypes.bool,
+
+  /**
+   * Provide custom text for the component for each translation id
+   */
+  translateWithId: PropTypes.func.isRequired,
+
+  /**
+   * `true` to use the mobile variant.
+   */
+  isMobile: PropTypes.bool,
+};
+
+NumberInputView.defaultProps = {
+  disabled: false,
+  hideLabel: false,
+  iconDescription: 'choose a number',
+  label: ' ',
+  onChange: () => {},
+  onClick: () => {},
+  step: 1,
+  value: 0,
+  invalid: false,
+  invalidText: 'Provide invalidText',
+  ariaLabel: 'Numeric input field with increment and decrement buttons',
+  helperText: '',
+  light: false,
```

Comment:
As long as we're making a new API can we switch to a `variant` approach and away from boolean props? Something like `variant="light"` to allow for future variations with props that don't conflict.